packages/core/src/external/aws/lambda-logic/bundle-to-html-derm.ts

```diff
@@ -0,0 +1,2688 @@
+  })?.code;
+
+  if (adminGenCode) {
+    return adminGenCode;
+  }
+
+  return null;
+}
+
+function getSpecificCode(coding: Coding[], systemsList: string[]): string | null {
+  // return the first code that matches the system
+  // systemList should be in order of priority
+
+  let specifiedCode: string | null = null;
+
+  if (systemsList.length) {
+    for (const system of systemsList) {
+      const code = coding.find(coding => {
+        return coding.system?.toLowerCase().includes(system);
+      })?.code;
+
+      if (code && !specifiedCode) {
+        specifiedCode = `${system.toUpperCase()}: ${code}`;
+      }
+    }
+  }
+
+  return specifiedCode;
+}
+
+function getConditionDatesFromEncounters(
+  encounters: Encounter[]
+): Record<string, { start: string; end: string }> {
+  const conditionDates: Record<string, { start: string; end: string }> = {};
+
+  encounters.forEach(encounter => {
+    if (encounter.diagnosis) {
+      encounter.diagnosis.forEach(diagnosis => {
+        if (diagnosis.condition && diagnosis.condition.reference) {
+          const conditionId = diagnosis.condition.reference.split("/")[1];
+          if (encounter.period && conditionId) {
+            conditionDates[conditionId] = {
+              start: encounter.period.start ?? "",
+              end: encounter.period.end ?? "",
+            };
+          }
+        }
+      });
+    }
+  });
+
+  return conditionDates;
+}
+
+function createSection(title: string, tableContents: string) {
+  return `
+    <div id="${title.toLowerCase().replace(/\s+/g, "-")}" class="section">
+      <div class="section-title">
+        <h3 id="${title}" title="${title}">&#x276F; ${title}</h3>
+        <a href="#mr-header">&#x25B2; Back to Top</a>
+      </div>
+      <div class="section-content">
+          ${tableContents}
+      </div>
+    </div>
+  `;
+}
+
+function mapResourceToId<ResourceType>(resources: Resource[]): Record<string, ResourceType> {
+  return resources?.reduce((acc, resource) => {
+    const id = resource?.id ?? "";
+
+    return {
+      ...acc,
+      [id]: resource,
+    };
+  }, {});
+}
+
+// find condition with code Z00 in the past year
+function getDermConditions(conditions: Condition[]): Condition[] {
+  const dermConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the L00-L99 range
+    const isDermCode = code ? /L\d{2}(?:\.\d+)?/i.test(code) : false;
+
+    return isDermCode;
+  });
+
+  return dermConditions;
+}
+
+function getRheumatoidConditions(conditions: Condition[]) {
+  const rheumatoidConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the M00-M99 range
+    const isRheumatoidCode = code ? /M\d{2}(?:\.\d+)?/i.test(code) : false;
+
+    return isRheumatoidCode;
```

Comment:
These are pretty specific to this MR type but when we get back to the engine i agree we can have them seperated

Reply:
Sounds good, but we could have them as functions on the same file - it would help us later to build the engine.

But NABD if you want to keep as is.

Reply:
You mean like the regex to be a specific function?

Reply:
No, create functions to check for "truthiness", like:

```ts
function isDermCondition(condition: Condition): boolean {
    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
    // Check if the code is in the L00-L99 range
    const isDermCode = code ? /L\d{2}(?:\.\d+)?/i.test(code) : false;
    return isDermCode;
}
```


and then we can use them like so:

```ts
const dermConditions = conditions.filter(isDermCondition);
```